ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/SniHandler.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.MISDIRECTED_REQUEST;
+
+import io.confluent.ksql.rest.Errors;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+public class SniHandler implements Handler<RoutingContext> {
```

Comment:
Add some quick unit tests for edge cases? That this handler only applies for TLS requests, that the port-handling works as expected, etc.

Ideally we'd add an integration test (and also check that the config sets up the new handler in the process) but that might be more trouble than it's worth.

Reply:
I'll add some unit tests